### PR TITLE
feat: configurable query for site URL

### PR DESCRIPTION
**Pull request by amclin**
The core Gatsby sitemap plugin allows for Configurable queries to get the siteUrl. See: https://www.gatsbyjs.org/packages/gatsby-plugin-sitemap

This is necessary because when using Gatsby to generate sites with dynamic URLs, you can't dynamically set the siteMetadata from a data source.

This PR adds the same capability to the robots.txt plugin so that it can behave consistently. If sitemap.xml is using a different domain, then robots.txt should be able to as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -194,6 +194,30 @@ module.exports = {
...
+
+`gatsby-config.js`
+
+```js
+module.exports = {
+  plugins: [
+    {
+      resolve: 'gatsby-plugin-robots-txt',
+      options: {
+        query: {
```
Comment (mdreizin) [inline README.md]: @amclin It looks like that we need to wrap `query` with quotes.

**Comment by mdreizin** [comment]
@amclin Have you had a chance to test 1.4.0 version?

**Comment by amclin** [comment]
Just tested in our project with 1.4.0. Working as intended. Thanks for the fast response and merge @mdreizin !

**Comment by mdreizin** [comment]
@amclin Thanks for contributing!

**Pull request merged**